### PR TITLE
ButtonComponent can restore purchases

### PR DESCRIPTION
**Pull request by JayShortway**
As the title says. I'm using the `PurchaseHandler` to handle the restore action, as I saw a similar approach being taken in https://github.com/RevenueCat/purchases-ios/pull/4332 for the purchase action.

### CODE REVIEW

File: Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/SamplePaywallsList.swift [hunk 1]
```diff
@@ -95,7 +95,7 @@ struct SamplePaywallsList: View {
                 paywallComponentsData: data,
                 offering: Self.loader.offeringWithDefaultPaywall(),
                 onDismiss: { self.display = nil }
-            )
+            ).environmentObject(Self.purchaseHandler)
```
Comment (JayShortway) [inline Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/SamplePaywallsList.swift:98]: I don't think this is how we wanna do it (permanently)? It's a separate commit, so I can easily revert it.

Comment (joshdholtz) [reply Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/SamplePaywallsList.swift:98]: Ah yeah, I fixed this is one of my packages commits! So if you want to remove this commit so the merge is clean that would be 🔥 

But thanks for calling this out 🫶

**Review by joshdholtz — APPROVED** [review]
This looks good!

If you want to remove that commit for adding the environment object in the PaywallsTester that would be 💯 since I fixed that in one of my packages PRs 💪 

Other than that this is good to merge!

File: Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/SamplePaywallsList.swift (hunk 1 above)
Comment (JayShortway) [reply Tests/TestingApps/PaywallsTester/PaywallsTester/UI/Views/SamplePaywallsList.swift]: Totally! Revert is here: https://github.com/RevenueCat/purchases-ios/pull/4372/commits/985b78a4aaa8b65e876ed990fc711786818b8ddc.

**Comment by JayShortway** [comment]
(@joshdholtz I see you reacted with a thumbs up, so I'm gonna go ahead and merge haha.)

**Pull request merged**